docs/content/manual/harvester-rancher/4678-load-balancer-different-roles-pool.md

```diff
@@ -0,0 +1,77 @@
+---
+title: Create load balancer on different roles pools in RKE2 cluster
+---
+
+* Related issues: [#4678](https://github.com/harvester/harvester/issues/4678) [BUG] Unable to successfully create LoadBalancer service from Guest cluster post 1.2.1 upgrade
+
+
+
+## Category: 
+* Rancher 
+
+## Verification Steps
+1. Prepare three nodes Harvester and import with Rancher
+1. Provision a RKE2 guest cluster in Rancher
+1. Create the first pool, specify the `etcd` and `control plane` roles
+1. Create the second pool, specify the `work` role only
+1. Add the following cloud config in the `User Data` of each pool
+    ```
+    write_files:
+    - encoding: b64
+    content: {harvester's kube config, the cluster namespace should be same as the pool you created (base64 encoded)}
+    owner: root:root
+    path: /etc/kubernetes/cloud-config
+    permission: '0644'
+    ```
+1. Get the Harvester kubeconfig file, remember to add the namespace (should be the same with the guest cluster)
+    ```
+    contexts:
+    - name: "local"
+    context:
+    user: "local"
+    cluster: "local"
+    namespace: "default" ---------------------> Add this line
+    ```
+1. Output the Harvester kubeconfig into base64 format without new line
+    ```
+    cat local.yaml | base64 -w 0
+    ```
+1. Copy the base64 encoded kubeconfig to the cloud config write file sections above
+{{< image "images/rancher/4678-base64-kubeconfig.png" >}}
+1. Provision the RKE2 guest cluster
+1. After pools are created, we remove the harvester-cloud-provider in Apps > Installed Apps (kube-system namespace).
+1. Add new charts in Apps > Repositories, use https://charts.harvesterhci.io to install and select 0.2.3.
+{{< image "images/rancher/4678-add-harvester-repo.png" >}}
+1. Install Harvester cloud provider 0.2.3 from market
+{{< image "images/rancher/4678-install-cloud-provider.png" >}}
+1. Create a nginx deployment using Harvester storage class
+1. Create a load balancer service with dhcp type and bind to nginx deployment
+1. Check can install the load balancer
+1. Create a standalone load balancer service, set the following annotation
+1. Access to the guest cluster VM, check the load balancer service can get `EXTERNAL_IP`
+
+1. Prepare three nodes Harvester and import with Rancher
+1. Provision a RKE2 guest cluster in Rancher
+1. Create the first pool, specif the `control plane` role only
+1. Create the second pool, specify the `etcd` role only
+1. Create the third pool, specify the `worker` role only
+1. Repeat the steps 13 - 17 to create load balancer service
+
+1. Prepare three nodes Harvester and import with Rancher
+1. Provision a RKE2 guest cluster in Rancher
+1. Create the only one pool, specif the control plane, etcd and workers roles
```

Comment:
This is already covered in automation test.

Reply:
After checking and confirmed with team, our existing backend e2e test of rancher integration in `test_9_rancher_integration.py`:

There we provide the ability to create 3 nodes guest cluster 
https://github.com/harvester/tests/blob/3d210686882bd73a548200138a6fedb7ea5d8067/harvester_e2e_tests/integrations/test_9_rancher_integration.py#L152

In actual testing helped by Albin, the 3 nodes downstream cluster are created in the same pool and all set to the `ALL` role.  
![image](https://github.com/harvester/tests/assets/29251855/79237452-2865-446f-81d3-9d4ba33cb2b4)

Thus we can assume this manual test plan to have different roles in different pools did not covered by e2e test.